include/xleaflet/xwidget_control.hpp

```diff
@@ -0,0 +1,132 @@
+/******************************************************************************
+* Copyright (c) 2018, Sylvain Corlay and Johan Mabille, Wolf Vollprecht and   *
+* Martin Renou                                                                *
+*                                                                             *
+* Distributed under the terms of the BSD 3-Clause License.                    *
+*                                                                             *
+* The full license is in the file LICENSE, distributed with this software.    *
+*******************************************************************************/
+
+#ifndef XLEAFLET_WIDGET_CONTROL_HPP
+#define XLEAFLET_WIDGET_CONTROL_HPP
+
+#include <string>
+
+#include "xwidgets/xholder.hpp"
+#include "xwidgets/xmaterialize.hpp"
+#include "xwidgets/xwidget.hpp"
+
+#include "xcontrol.hpp"
+
+namespace xlf
+{
+    /******************************
+     * widget_control declaration *
+     ******************************/
+
+    template <class D>
+    class xwidget_control : public xcontrol<D>
+    {
+    public:
+
+        using base_type = xcontrol<D>;
+        using derived_type = D;
+
+        using widget_type = xw::xholder<xw::xwidget>;
+
+        void serialize_state(xeus::xjson&, xeus::buffer_sequence&) const;
+        void apply_patch(const xeus::xjson&, const xeus::buffer_sequence&);
+
+        XPROPERTY(widget_type, derived_type, widget);
+        XPROPERTY(xtl::xoptional<int>, derived_type, max_width);
+        XPROPERTY(xtl::xoptional<int>, derived_type, min_width);
+        XPROPERTY(xtl::xoptional<int>, derived_type, max_height);
+        XPROPERTY(xtl::xoptional<int>, derived_type, min_height);
+
+    protected:
+
+        xwidget_control();
+        using base_type::base_type;
+
+    private:
+
+        void set_defaults();
+    };
+
+    using widget_control = xw::xmaterialize<xwidget_control>;
+
+    using widget_control_generator = xw::xgenerator<xwidget_control>;
+
+    /*************************
+     * xwidget_control implementation *
+     *************************/
```

Comment:
You are still missing stars, sorry I did not put a comment here as well the first time

Reply:
I like your sense of details

Reply:
details are important ;)